netlink-packet-netfilter/src/nflog/config.rs

```diff
@@ -0,0 +1,160 @@
+use anyhow::Context;
+use byteorder::{BigEndian, ByteOrder};
+use netlink_packet_core::{
+    DecodeError,
+    NetlinkHeader,
+    NetlinkMessage,
+    NetlinkPayload,
+    NLM_F_ACK,
+    NLM_F_REQUEST,
+};
+use netlink_packet_utils::{
+    nla::{DefaultNla, Nla, NlaBuffer},
+    parsers::{parse_u16_be, parse_u32_be, parse_u8},
+    Parseable,
+};
+
+pub use config_cmd::ConfigCmd;
+pub use config_flags::ConfigFlags;
+pub use config_mode::{ConfigMode, CopyMode};
+pub use timeout::Timeout;
+
+use crate::{
+    constants::NFNETLINK_V0,
+    message::{NetfilterHeader, NetfilterMessage},
+};
+
+use super::NfLogMessage;
+
+mod config_cmd;
+mod config_flags;
+mod config_mode;
+mod timeout;
+
+pub const NFULA_CFG_CMD: u16 = libc::NFULA_CFG_CMD as u16;
+pub const NFULA_CFG_MODE: u16 = libc::NFULA_CFG_MODE as u16;
+pub const NFULA_CFG_NLBUFSIZ: u16 = libc::NFULA_CFG_NLBUFSIZ as u16;
+pub const NFULA_CFG_TIMEOUT: u16 = libc::NFULA_CFG_TIMEOUT as u16;
+pub const NFULA_CFG_QTHRESH: u16 = libc::NFULA_CFG_QTHRESH as u16;
+pub const NFULA_CFG_FLAGS: u16 = libc::NFULA_CFG_FLAGS as u16;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ConfigNlas {
+    Cmd(ConfigCmd),
+    Mode(ConfigMode),
+    NlBufSiz(u32),
+    Timeout(Timeout),
+    QThresh(u32),
+    Flags(ConfigFlags),
+    Other(DefaultNla),
+}
+
+impl From<ConfigCmd> for ConfigNlas {
```

Comment:
We could maybe automate this with a small macro.

Reply:
Adding a macro for that seems overkill. But if you want to go this way, what do you think about using the derive-more crate for that?

Reply:
Ah yes I like `derive_more`.